dfusion_rust_core/src/models/solution.rs

```diff
@@ -3,24 +3,35 @@ use super::*;
 use log::info;
 
 use byteorder::{BigEndian, ByteOrder};
+use std::collections::HashMap;
 use std::iter::once;
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct Solution {
-    pub prices: Vec<u128>,
+    /// token_id => price
+    pub prices: HashMap<u16, u128>,
     pub executed_buy_amounts: Vec<u128>,
     pub executed_sell_amounts: Vec<u128>,
 }
 
 impl Solution {
     pub fn trivial(num_orders: usize) -> Self {
         Solution {
-            prices: vec![],
+            prices: HashMap::new(),
             executed_buy_amounts: vec![0; num_orders],
             executed_sell_amounts: vec![0; num_orders],
         }
     }
 
+    /// Returns the price for a token by ID or 0 if the token was not found.
+    pub fn price(&self, token_id: u16) -> Option<u128> {
+        self.prices.get(&token_id).cloned()
```

Comment:
ubernit: I would suggest to use `copied()` for consistency with the `max_token` function and to give signal that the `Option<&T> -> Option<T>` is computationally inexpensive.